### DISCUSSION TITLE
"New-MockObject" adding multiple methods - leads to error!?

### DISCUSSION THREAD

**Discussion by ghost** [pester/Pester#2066] [Q&A] [answered]
Hello Together,
sorry if the question is stupid, I am still new to Pester.
I am trying to mock an object with multiple methods, which results in an error.
Here is the code snippet directly from powershell:
```powershell
$test = New-MockObject -Type 'Net.Sockets.TcpClient' -Methods  @{
    Connect = { param($Server, $Port)"connect" };
    Close   = { param($Server, $Port)"close" }
}
```
Error Message:
```powershell
InvalidOperation: C:\Program Files\PowerShell\Modules\Pester\5.3.0\Pester.psm1:12499:13
 Line |
12499 |              $mockState[$historyName] = @{
      |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | Unable to index into an object of type "System.Management.Automation.PSNoteProperty".
```

Works fine with single method:
```powershell
$test = New-MockObject -Type 'Net.Sockets.TcpClient' -Methods  @{
    Connect = { param($Server, $Port)"connect" }
}
```

Also fine with multiple properties:
```powershell
$test = New-MockObject -Type 'Net.Sockets.TcpClient' -Properties  @{
    Connect = "connect" ;
    Close   = "close"
}
```
Using Pester Version 5.3.0 from psgallery on Windows Powershell Core 7.1.4

Thanks already for feedback

**Comment by nohwnd** [comment, marked as answer]
Not a stupid question. It was a bug https://github.com/pester/Pester/pull/2068

**Reply by ghost** [reply]
Great, thanks for the fast response. Next time I am going to have a better search.

**Reply by nohwnd** [reply]
I created the bug from your question, so you could not find it. It is a new feature, and I made a mistake in the implementation. :)
